### PR TITLE
Salt minion improvements

### PR DESCRIPTION
**Pull request by aneeshusa**
###### Motivation for this change
Review commit by commit.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/admin/salt/minion.nix [hunk 1]
```diff
@@ -28,24 +27,33 @@ in
...
     systemd.services.salt-minion = {
       description = "Salt Minion";
       wantedBy = [ "multi-user.target" ];
       after = [ "network.target" ];
-      path = with pkgs; [
-        utillinux
-      ];
+      path = [
+        (builtins.dirOf config.security.wrapperDir)
+      ] ++ config.environment.profiles;
```
Comment (Infinisil) [inline nixos/modules/services/admin/salt/minion.nix]: I haven't seen `environment.profiles` being used like this, but it seems like a good solution

**Review by infinisil — APPROVED** [review]

File: nixos/modules/services/admin/salt/minion.nix (hunk 1 above)
Comment (andir) [reply nixos/modules/services/admin/salt/minion.nix]: One thing:
You might want to consider reversing the list. The way this list seems to work is that the most specific thing comes last. Thats why reversing it when creating a `PATH` might be a good idea. 

See: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/environment.nix#L58

**Comment by Infinisil** [comment]
@aneeshusa Can you do the thing mentioned by @andir? I can then merge it

**Comment by Infinisil** [comment]
Hmm, I'm questioning this way of adding to PATH now. A user could also just set `systemd.services.salt-minion.path = config.environment.profiles` in their own module to get access to all installed program. It seems to be a source of potential non-reproducibility. Adding something to `environment.systemPackages` (which happens a lot in my system) could potentially alter binaries and stuff, which can have an effect on salt-minion, which seems weird. What's your motivation for doing it this way?

Maybe it would be better to just mention in the option description how users can add some packages to PATH for salt-minion (via `systemd.services.salt-minion.path`).

**Comment by aneeshusa** [comment]
Those are reasonable concerns, so going to close this. The other two useful commits are now included in #48499.

**Pull request closed without merging**